### PR TITLE
fix for #2269 - preview for uploaded documents

### PR DESCRIPTION
**Pull request by aks681**
I added a preview button to documents tab of viewclient.html and in viewloanaccountdetails.html. I tested it on images (both jpg and png) and it works.

### CODE REVIEW

**Comment by mbj36** [comment]
Thanks for the PR, It will be reviewed Shortly!

This message was posted by [GitMate.io](https://gitmate.io) with timestamp signature `Thanks for the PR, It will be reviewed Shortly!:1eMWxw:8u-78dprugO_-eFUInH1HvDuPts`

**Comment by gkrishnan724** [comment]
@aks681 When I click on the preview button, It is downloading the document for me.

**Comment by gkrishnan724** [comment]
@aks681 It is working fine for me, but It would be better If you could somehow make it preview on the same page it will look much neater.

**Comment by aks681** [comment]
Yes, the first time I am previewing it, it downloads but then on, it previews normally.
You mean same page, below the table?

**Comment by gkrishnan724** [comment]
Yeah Either below the table or to a new route. 

Take a look at  Transaction bills. I think you will understand.

**Comment by aks681** [comment]
Ok, will try that.

**Comment by gkrishnan724** [comment]
@aks681 Do you have an Idea why it downloaded the first time? Since it happened to me too but It worked fine afterward. I thought I might have clicked the wrong button. If you could fix that too It would be great!

**Comment by aks681** [comment]
I am not sure about the auto download, will test it again. Is this preview enough?

File: app/views/clients/viewclient.html [hunk 1]
```diff
@@ -965,11 +965,13 @@ <h4><strong>{{client.displayName}}</strong></h4>
 							<td>{{clientdocument.fileName}}</td>
 							<td>
 								<a target="_blank" href="{{hostUrl}}{{clientdocument.docUrl}}"><i class="fa fa-cloud-download"></i></a>&nbsp;
+								<button class="btn btn-default" ng-click="previewDocument(clientdocument.docUrl,clientdocument.fileName)"> Preview </button>
 								<a ng-click="deleteDocument(clientdocument.id,$index)"><i class="fa fa-times"></i></a>
 							</td>
 						</tr>
 						</tbody>
 					</table>
+					<object height="100%" width="100%" ng-show="preview" data="{{fileUrl}}" type="{{fileType}}"></object>
```
Comment (gkrishnan724) [inline app/views/clients/viewclient.html]: please make this a fixed width and height.

File: app/views/loans/viewloanaccountdetails.html [hunk 2]
```diff
@@ -888,11 +888,13 @@ <h3>{{'label.heading.currentBalance' | translate}}: {{loandetails.currency.displ
                                 <td>{{document.fileName}}</td>
                                 <td>
                                     <a target="_blank" href="{{hostUrl}}{{document.docUrl}}"><i class="fa fa-cloud-download"></i></a>&nbsp;
+                                    <button class="btn btn-default" ng-click="previewDocument(document.docUrl,document.fileName)"> Preview </button>
                                     <a ng-click="deleteDocument(document.id,$index)"><i class="fa fa-times"></i></a>
                                 </td>
                             </tr>
                             </tbody>
                         </table>
+                        <object height="100%" width="100%" ng-show="preview" data="{{fileUrl}}" type="{{fileType}}"></object>
```
Comment (gkrishnan724) [inline app/views/loans/viewloanaccountdetails.html]: this too.

**Comment by gkrishnan724** [comment]
@aks681  PDF files are not working :( Tried with two PDF's. Getting the error cannot Load PDF. 

Also, please replace the button with an appropriate Icon with a tooltip would look neat.  You could use this icon: http://fontawesome.io/icon/eye/

File: app/views/clients/viewclient.html (hunk 1 above)
Comment (aks681) [reply app/views/clients/viewclient.html]: Ok, what values should I give?

**Comment by aks681** [comment]
Can you tell which user has pdf files uploaded so I can test it?

**Comment by aks681** [comment]
I think the response type at the server side has to be changed for it to work with pdfs, tried a lot of methods, either it auto-downloaded or it showed failed to load pdf.

**Comment by gkrishnan724** [comment]
@aks Okay, I guess we can keep it for Image files then.

**Comment by aks681** [comment]
Ok, Is there anything else to add?

File: app/views/clients/viewclient.html [hunk 3]
```diff
@@ -965,11 +965,13 @@ <h4><strong>{{client.displayName}}</strong></h4>
 							<td>{{clientdocument.fileName}}</td>
 							<td>
 								<a target="_blank" href="{{hostUrl}}{{clientdocument.docUrl}}"><i class="fa fa-cloud-download"></i></a>&nbsp;
+								<a ng-click="previewDocument(clientdocument.docUrl,clientdocument.fileName)"><i class="fa fa-eye"></i></a>
```
Comment (gkrishnan724) [inline app/views/clients/viewclient.html]: The icon should only be visible if the filetype is an Image.

**Comment by gkrishnan724** [comment]
Also, just squash the commits into one and we should be good to go

File: app/views/clients/viewclient.html (hunk 3 above)
Comment (aks681) [reply app/views/clients/viewclient.html]: Ok, will add that.

**Comment by aks681** [comment]
Made the changes and squashed into one commit.

File: app/views/clients/viewclient.html [hunk 4]
```diff
@@ -965,11 +965,13 @@ <h4><strong>{{client.displayName}}</strong></h4>
 							<td>{{clientdocument.fileName}}</td>
 							<td>
 								<a target="_blank" href="{{hostUrl}}{{clientdocument.docUrl}}"><i class="fa fa-cloud-download"></i></a>&nbsp;
+								<a ng-show="clientdocument.fileIsImage" ng-click="previewDocument(clientdocument.docUrl,clientdocument.fileName)"><i class="fa fa-eye"></i></a>
 								<a ng-click="deleteDocument(clientdocument.id,$index)"><i class="fa fa-times"></i></a>
 							</td>
 						</tr>
 						</tbody>
 					</table>
+					<object height="400px" width="400px" ng-show="preview" data="{{fileUrl}}" type="{{fileType}}"></object>
```
Comment (gkrishnan724) [inline app/views/clients/viewclient.html]: Sorry about making you do changes again and again, I forgot to tell about making this center aligned my bad. Also the Image size could be width="70%" height="60%" It looked kind of better to me.

I noticed that `.jpeg` files don't have preview so I think they should be included too.

Sorry for the late comments.

You could make the changes then do 

1. `git add` - Add changes
2. `git commit --amend` - Overwrite previous commit
3. `git push -f origin upload-preview

Comment (aks681) [reply app/views/clients/viewclient.html]: Ok, made the changes. Also align:center didn't work, so I used padding-left: 25% for center align.

**Comment by gkrishnan724** [comment]
Thanks :)

**Pull request merged**